### PR TITLE
Rename connection to tunnel

### PR DESCRIPTION
**Pull request by alalamav**
- Merges #863 into the go-tun2socks branch.
- Resolved merge conflicts in:
  - apple/xcode/ios/Outline.xcodeproj/project.pbxproj
  - apple/xcode/osx/Outline.xcodeproj/project.pbxproj
  - cordova-plugin-outline/apple/vpn/PacketTunnelProvider.m
  -  cordova-plugin-outline/android/java/org/outline/OutlinePlugin.java
  - cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnelService.java
  -  src/electron/process_manager.ts
- Tested that the change builds and runs in all platforms.

### CODE REVIEW

**Review by JonathanDCohen — COMMENTED** [review]
I ran `grep -n '[Cc]onnection' cordova-plugin-outline/**/*` and got

```
cordova-plugin-outline/apple/src/OutlineVpn.swift:179:      let session = self.tunnelManager?.connection as! NETunnelProviderSession
cordova-plugin-outline/apple/src/OutlineVpn.swift:190:    let session: NETunnelProviderSession = tunnelManager?.connection as! NETunnelProviderSession
cordova-plugin-outline/apple/src/OutlineVpn.swift:301:    let vpnStatus = tunnelManager?.connection.status
cordova-plugin-outline/apple/src/OutlineVpn.swift:309:                                              object: manager.connection)
cordova-plugin-outline/apple/src/OutlineVpn.swift:311:                                           name: .NEVPNStatusDidChange, object: manager.connection)
cordova-plugin-outline/apple/src/OutlineVpn.swift:317:    if let vpnStatus = tunnelManager?.connection.status {
cordova-plugin-outline/apple/src/OutlineVpn.swift:363:    let session: NETunnelProviderSession = tunnelManager?.connection as! NETunnelProviderSession
grep: cordova-plugin-outline/apple/src/macos: Is a directory
grep: cordova-plugin-outline/apple/vpn: Is a directory
cordova-plugin-outline/apple/vpn/OutlineTunnelStore.swift:22:  // TODO(alalama): s/connection/tunnel when we update the schema.
cordova-plugin-outline/apple/vpn/OutlineTunnelStore.swift:23:  private static let kTunnelStoreKey = "connectionStore"
cordova-plugin-outline/apple/vpn/OutlineTunnelStore.swift:24:  private static let kTunnelStatusKey = "connectionStatus"
cordova-plugin-outline/apple/vpn/PacketTunnelProvider.m:141:                                                 code:NEVPNErrorConnectionFailed
cordova-plugin-outline/apple/vpn/PacketTunnelProvider.m:156:                                                            code:NEVPNErrorConnectionFailed
cordova-plugin-outline/apple/vpn/PacketTunnelProvider.m:503:                                                    code:NEVPNErrorConnectionFailed
cordova-plugin-outline/apple/vpn/PacketTunnelProvider.m:512:                                                    code:NEVPNErrorConnectionFailed
```

I also ran `find src -type f | grep -v messages | grep -v licenses | xargs grep -n '[Cc]onnection'` and found

```
src/types/modules/socks/index.d.ts:34:  export function createConnection(
src/www/app/electron_main.ts:25:import {FakeOutlineTunnel} from './fake_connection';
src/www/app/fake_connection.ts:19:// Note that because this implementation does not emit disconnection events, "switching" between
src/www/app/app.ts:214:      console.warn('server card not found after disconnection event, assuming forgotten');
src/www/app/app.ts:400:            this.errorReporter.report(`connection failure: ${e.name}`, 'connection-failure');
src/www/app/outline_server.ts:77:      // TODO: None of the plugins currently return an ErrorCode on disconnection.
src/www/app/cordova_main.ts:25:import {FakeOutlineTunnel} from './fake_connection';
src/www/ui_components/server-card.html:26:<link rel="import" href="server-connection-viz.html" />
src/www/ui_components/server-card.html:67:      .card-header server-connection-viz {
src/www/ui_components/server-card.html:151:        <server-connection-viz
src/www/ui_components/server-card.html:155:        ></server-connection-viz>
src/www/ui_components/server-card.html:181:            <server-connection-viz state="[[state]]" root-path="[[rootPath]]" expanded></server-connection-viz>
src/www/ui_components/server-connection-viz.html:18:<dom-module id="server-connection-viz">
src/www/ui_components/server-connection-viz.html:248:      is: "server-connection-viz",
src/www/ui_components/servers-view.html:18:<link rel="import" href="server-connection-viz.html" />
src/www/ui_components/servers-view.html:89:            <server-connection-viz state="ZERO_STATE" root-path="[[rootPath]]" expanded></server-connection-viz>
src/www/ui_components/feedback-view.html:81:            <paper-item value="connection">[[localize('feedback-connection')]]</paper-item>
src/electron/find_tap_device_name.bat:56:  set ADAPTER_CONFIG_KEY=!NET_CONFIG_KEY!\!ADAPTER_NET_CONFIG_ID!\Connection
src/electron/add_tap_device.bat:88:::   This network connection does not exist.
src/electron/tunnel_store.ts:35:    this.storagePath = path.join(storagePath, 'connection_store');
src/electron/connectivity.ts:21:// Resolves iff a (TCP) connection can be established with the specified destination within the
src/electron/routing_service.ts:15:import {createConnection, Socket} from 'net';
src/electron/routing_service.ts:37:  connectionStatus: TunnelStatus;
src/electron/routing_service.ts:79:  // Fulfills once a connection is established with the routing daemon *and* it has successfully
src/electron/routing_service.ts:83:      const newSocket = this.socket = createConnection(SERVICE_NAME, () => {
src/electron/routing_service.ts:145:          this.networkChangeListener(message.connectionStatus);
src/electron/routing_service.ts:175:  // Use #onceDisconnected to be notified when the connection terminates.
```

It seems the ones in cordova-plugin outline are more relevant but I wanted to flag these to you to double check

File: cordova-plugin-outline/README.md [hunk 1]
```diff
@@ -44,17 +44,17 @@ We use a custom framework dependency, PacketProcessor, that implements tun2socks
...
-This plugin targets Apple devices running iOS 9.0+ and macOS/OS X 10.11.
+This plugin targets Apple devices running iOS 11.0+ and macOS/OS X 10.11.
 
 ## JavaScript API
 
 ```ts
 
 // Represents a Shadowsocks server configuration.
-interface ShadowsocksServerConfig {
+interface ServerConfig {
```
[the remark this answers is not in the document]
Comment (alalamav) [reply cordova-plugin-outline/README.md:57]: We have file paths in the source definitions, see https://github.com/Jigsaw-Code/outline-client/blob/master/cordova-plugin-outline/outlinePlugin.js#L42

**Comment by alalamav** [comment]
Thanks for posting the diff. It looks like I missed renaming the `fake_connection.ts` file, but the source was renamed. It should be fine to handle that on a separate PR in master. Other than that, I didn't spot anything worrisome. LMK what you think.

**Comment by JonathanDCohen** [comment]
Separate PR in master should be fine

**Review by JonathanDCohen — APPROVED** [review]

**Pull request merged**